app/controllers/buyers/accounts_controller.rb

```diff
@@ -52,6 +52,7 @@ def create
       redirect_to admin_buyers_account_path(@buyer)
     else
       @user = signup_result.user
+      flash[:error] = signup_result.errors.messages.without(:account, :user).values.join(', ')
```

Comment:
Flash is stored into cookies to persist across redirect unless used.
It is better to use `flash.now` if you know it will be used in the same request